### DISCUSSION TITLE
Improving Result Relevance for Documents with Minimal Information

### DISCUSSION THREAD

**Discussion by underthesand** [meilisearch/product#746] [Feedback & Feature Proposal]
Hello Meilisearch team,

I'm currently experiencing an issue where documents with minimal information are being highlighted in search results, especially when there aren't many highly relevant results available. This leads to a suboptimal user experience as less informative documents appear more prominently than they should.

To address this, I've been exploring ways to adjust the ranking rules and boost the relevance of more informative documents. 
I've introduced a custom ranking attribute, `documentDensity`, which measures the amount of information in each document. 

Here is an example of the configuration I’m using:

```python
index.update_ranking_rules([
    "words",
    "typo",
    "proximity",
    "attribute",
    "sort",
    "exactness",
    "documentDensity:desc"
])
```

Despite these efforts, I'm still facing challenges in consistently improving the relevance of the search results. Could you please provide any best practices or additional strategies to address this issue? Specifically, I'm looking for advice on:

- Effective ways to calculate and use an `information_density` attribute or similar metric.
- Optimal ranking rule configurations to balance relevance and information richness.
- Any other approaches that might help improve the overall quality of search results.

Thank you for your assistance and for developing such a powerful search engine. Looking forward to your guidance.

**Comment by ManyTheFish** [comment]
Hello @underthesand,
If I understand correctly, your goal is to favor the documents specific to the query over the generalistic documents that could be related to it. Am I right?
Can you explain how you compute your `information_density` for your documents?

A first suggestion we could make is to move the `documentDensity:desc` upper in the ranking rule list, giving it more weight in the search result, maybe between `proximity` and `attribute`. 🤔
Moreover, using the [ranking score detail](https://www.meilisearch.com/docs/reference/api/search#ranking-score) may help you to rerank the results on your side to better fit your relevancy expectations.

Thank you for opening this discussion.
See you!
